drivers/core_demographics_report/app/models/core_demographics_report/household_type_calculations.rb

```diff
@@ -60,38 +60,78 @@ def household_type_client_percentage(type)
     def household_type_data_for_export(rows)
       rows['_Household Types'] ||= []
       rows['*Household Types'] ||= []
-      rows['*Household Types'] += ['Household Type', nil, 'Count', 'Percentage', nil]
+      rows['*Household Types'] += ['Household Type', nil, 'Client Count', 'Household Count', 'Percentage of Household', nil]
+      available_coc_codes.each do |coc_code|
+        rows['*Household Types'] += ["#{coc_code} Client"]
+        rows['*Household Types'] += ["#{coc_code} Household"]
+        rows['*Household Types'] += ["#{coc_code} Percentage"]
+      end
+      rows['*Household Types'] += [nil]
       available_household_types.invert.each do |id, title|
         rows["_Household Types_data_#{title}"] ||= []
         rows["_Household Types_data_#{title}"] += [
           title,
           nil,
+          household_type_client_count(id),
           household_type_hoh_count(id),
           household_type_hoh_percentage(id) / 100,
+          nil,
         ]
+        available_coc_codes.each do |coc_code|
+          rows["_Household Types_data_#{title}"] += [
+            household_type_hoh_count(id, coc_code.to_sym),
+            household_type_hoh_count(id, coc_code.to_sym),
+            household_type_hoh_percentage(id, coc_code.to_sym) / 100,
+          ]
+        end
       end
       rows
     end
 
-    private def enrollment_ids_in_household_type(key)
-      client_households[key]&.values&.flatten
+    private def enrollment_ids_in_household_type(key, coc_code = base_count_sym)
+      client_households(coc_code)[key]&.values&.flatten
     end
 
-    private def hoh_enrollment_ids_in_household_type(key)
-      hoh_households[key]&.values&.flatten
+    private def hoh_enrollment_ids_in_household_type(key, coc_code = base_count_sym)
+      hoh_households(coc_code)[key]&.values&.flatten
     end
 
     def available_household_types
       @filter.available_household_types.merge('Youth Only' => :unaccompanied_youth, 'Unknown' => :unknown)
     end
 
     private def calculate_households
+      @hoh_enrollments ||= Rails.cache.fetch([self.class.name, cache_slug, "#{__method__}_hoh_enrollments"], expires_in: expiration_length)
+      @households ||= Rails.cache.fetch([self.class.name, cache_slug, "#{__method__}_households"], expires_in: expiration_length)
+
+      return unless @hoh_enrollments.nil? && @households.nil?
```

Comment:
Hmm, this isn't how we usually use Rails.cache.fetch.  It' might work just fine, and maybe this is the right way since you're setting two instance variables that use related calculations.  Usually we pass a block to cache.fetch, which is only executed if the cached variable doesn't exist.

Another, perhaps bigger issue, is that we don't want to unintentionally change the Core Demographics report, which uses these same concerns.  I think we need to move `available_coc_codes` into an attribute accessor and set it from the calling controller.  That way we can ensure we never set it to anything but an empty array for Core Demographics, but can pass the appropriate CoC Codes for the Demographic Summary report.

Reply:
I had a thought on how to prevent running the CoC Calculations when unnecessary, I'll push some code shortly.

Reply:
I added the caching in for the hoh_enrollments and households variables because the query setting them was a long running one for this section and it helped with development. I'm putting together a new commit with the other suggestions and will plan on pulling it back out to keep the caching consistant with the rest of the application.